packages/context/src/interceptor-chain.ts

```diff
@@ -0,0 +1,222 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as debugFactory from 'debug';
+import {BindingFilter} from './binding-filter';
+import {BindingAddress} from './binding-key';
+import {BindingComparator} from './binding-sorter';
+import {Context} from './context';
+import {transformValueOrPromise, ValueOrPromise} from './value-promise';
+
+const debug = debugFactory('loopback:context:interceptor-chain');
+
+/**
+ * Return value for a method invocation
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type InvocationResult = any;
```

Comment:
I find it weird to see type name `InvocationResult` alongside `GenericInterceptorChain`, "invocation" seems like a specialization of "interceptor"  to me.

Lets' find a better name please, e.g. `InterceptorResult` or perhaps `NextResult`/`NextReturnValue`.

When I think about this more, maybe `GenericInterceptor` should accept the "result" type as a generic argument too, allowing specializations to provide better type information where possible? (Of course, invocation interceptor will use `any`.)

Reply:
Added a new type param `T`.